include/SDL3/SDL_begin_code.h

```diff
@@ -173,6 +173,19 @@
 #endif /* C++17 or C2x */
 #endif /* SDL_FALLTHROUGH not defined */
 
+#ifndef SDL_NODISCARD
+#if (defined(__cplusplus) && __cplusplus >= 201703L) || \
+    (defined(__STDC_VERSION__) && __STDC_VERSION__ >= 202311L)
+#define SDL_NODISCARD [[nodiscard]]
+#elif ( (defined(__GNUC__) && (__GNUC__ >= 4)) || defined(__clang__) )
+#define SDL_NODISCARD __attribute__((warn_unused_result))
+#elif defined(_MSC_VER)
+#define SDL_NODISCARD _Check_return_
```

Comment:
is there an MSC version requirement for this?

Reply:
After googling a bit it seems that the requirement is `_MSC_VER >= 1700`, will add it.